File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        //if negative: return false
        if (x<0){
            return false;
        }
        int copyX = x;
        int reverted = 0;
        while(x>0){
            reverted = reverted*10 + x%10;
            x /= 10;
        }
        return reverted == copyX;
    }
}

File: palindrome-number.cpp
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0) return false;;
        int copy = x;
        int revert = 0;
        while(x>0){
            revert = revert*10 + x%10;
            x/=10;
        }
        return revert == copy;
    }
}